### ISSUE TITLE
rustc panic: cannot convert `ReScope(Node(265))` to a region vid

### ISSUE THREAD

**Issue by nlevitt** [rust-lang/rust#61951]
rustc panic, in a nutshell:
```
error: internal compiler error: src/librustc_mir/borrow_check/nll/universal_regions.rs:741: cannot convert `ReScope(Node(265))` to a region vid
```

Full output: [compiler-panic.txt](https://github.com/rust-lang/rust/files/3305185/compiler-panic.txt)

Reproducible for me like so:

1. git clone git@github.com:nlevitt/monie.git
2. git checkout 2342a831c50eee
3. apply this trivial patch (without this change, the program builds and runs fine)
```
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -122,7 +122,7 @@ fn proxy_request<T: Mitm + Sync + Send + 'static>(
                     let response = mitm2.response_headers(response);
                     let (parts, body) = response.into_parts();
                     let body =
-                        Body::wrap_stream(body.map(move |chunk| {
+                        Body::wrap_stream(body.map(|chunk| {
                             mitm2.response_body_chunk(chunk)
                         }));
                     Response::from_parts(parts, body)
```
4. cargo build

Rust version info:
```
$ rustc --verbose --version
rustc 1.35.0 (3c235d560 2019-05-20)
binary: rustc
commit-hash: 3c235d5600393dfe6c36eeed34042efad8d4f26e
commit-date: 2019-05-20
host: x86_64-apple-darwin
release: 1.35.0
LLVM version: 8.0
```

**Comment by matthewjasper** [comment]
Fixed in beta, and the release team has decided against a point release just for this.

**Comment by TjeuKayim** [comment]
I also experienced this bug. It was resolved after updating from 1.35.0 to 1.36.0.
